### PR TITLE
AF: vol-12x + eval-every-3 + seed=42 (seed variation)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
Same as violet's config (vol-12x + eval-every-3) but seed=42 instead of default. AF has shown seed sensitivity (#3254) — seed=42 may find a different basin where both metrics beat baseline. Programme-best surface was at vol-12x; eval-3 adds more epochs.

## Instructions
```bash
cd target/icml2026 && python train.py --dataset airfrans --optimizer lion --lr 1.25e-4 \
  --cosine-t-max 50 --grad-clip 1.0 --weight-decay 1e-2 \
  --model-layers 4 --model-hidden-dim 256 --model-heads 4 --enable-fourier \
  --epochs 999 --ema-decay 0.999 --use-lookahead --compile-model \
  --vol-loss-weight 12 --eval-interval 3 --save-checkpoint --seed 42 \
  --wandb-group af-eval-every-3-vol-sweep
```
**Target:** surface_mse < 0.000296 AND vol_mse < 0.002039

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT tanjiro:

## Results

### AF: vol-12x + eval-every-3 + seed=42

| Metric | Baseline | This Run (best) | Epoch | vs Baseline |
|--------|----------|-----------------|-------|-------------|
| surface_mse | 0.000296 | 0.000730 | 354 | +146.6% ❌ |
| vol_mse | 0.002039 | 0.003568 | 354 | +75.0% ❌ |

**Neither metric beats baseline.**

### Command
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py --dataset airfrans --optimizer lion --lr 1.25e-4 \
  --cosine-t-max 50 --grad-clip 1.0 --weight-decay 1e-2 \
  --model-layers 4 --model-hidden-dim 256 --model-heads 4 --enable-fourier \
  --epochs 999 --ema-decay 0.999 --use-lookahead --compile-model \
  --volume-loss-weight 12 --eval-interval 3 --save-checkpoint --seed 42 \
  --wandb-group af-eval-every-3-vol-sweep --agent tanjiro --wandb-name "tanjiro/af-vol12x-eval3-seed42"
```

Note: PR instructions used `--vol-loss-weight` but the actual CLI flag is `--volume-loss-weight`. Corrected for the run.

### W&B Run
- **Run ID:** ot10h2qx
- **Peak VRAM:** ~83.6 GB (GPU 0)
- **Epochs completed:** 409 (killed after divergence)

### What happened

The model trained stably through epoch 364 (loss ~0.1, grad_norm ~10-15), reaching its best surface_mse=0.000730 at epoch 354. Then at epoch 365, gradient norms spiked to 73.6 and training diverged catastrophically:

- **ep364:** loss=0.17, gnorm=19.0 (stable)
- **ep365:** loss=0.37, gnorm=73.6 (spike)
- **ep370:** loss=0.46, gnorm=52.7 (escalating)
- **ep389:** loss=76.3 (exploded)
- **ep407:** grad_norm=Infinity, grad_nonfinite=321/360
- **ep408:** loss=47.1, surface_mse=1.39 (fully diverged)

The divergence coincides with the LR cosine peak (T_max=50, epoch 350 = 7th cycle peak). Lion optimizer at lr=1.25e-4 appears too aggressive for long training — the repeated LR peaks eventually trigger instability. The model never recovered after the gradient explosion.

Best results at epoch 354 (surface_mse=0.000730, vol_mse=0.003568) are 2.5x and 1.75x worse than baseline respectively. The 4-layer Lion config explores a different region of loss landscape vs the 2-layer AdamW baseline, but doesn't converge as deeply.

### Suggested follow-ups

- **Lower Lion LR:** Reduce to 5e-5 or 7.5e-5 to prevent late-training instability at LR cosine peaks
- **LR warmup + decay:** Use cosine decay without restart (larger T_max = total epochs) to avoid repeated peaks triggering instability
- **Gradient norm monitoring + early stopping on divergence:** Could save GPU time by detecting the grad_norm spike pattern and stopping before full explosion
- **2-layer config with seed=42:** The baseline's 2-layer AdamW architecture may be more stable for AF — try seed=42 with the proven 2L config instead of 4L Lion

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #3343 because ADVISOR: Closing — same systematic config error as #3338 and #3355. Lion 4L/256d is not competitive on AF; the champion is AdamW 2L/256d. Surface 2.5x worse, volume 1.75x worse, diverged at ep365.

**IMPORTANT finding from your report:** You confirm that `--volume-loss-weight` IS the correct CLI flag (not `--vol-loss-weight` as I originally wrote in some PR instructions). Thank you for catching this — it means the vol-weight was applied correctly in these runs, and the poor results are purely due to the Lion 4L architecture mismatch, not a flag issue.

I'll reassign you to test eval-every-3 with the CORRECT AF champion config (AdamW 2L/256d)..
